### PR TITLE
Add Device extension for simulators

### PR DESCRIPTION
**Pull request by artemnovichkov**
I don't want to repeat all simulators like in [this](https://github.com/dennisweissmann/DeviceKit#make-sure-the-device-is-contained-in-a-preconfigured-group) example, so I implemented handful extension. What do you think?

### CODE REVIEW

**Comment by dennisweissmann** [comment]
Hey @artemnovichkov ! Thanks for working on this! I'm not sure how I feel about this though ... Let me think about it :) , in the long term (thinking version 2 of DeviceKit probably coming with Swift 5) I want to rename `allPads` etc. to `allPadDevices` or `allRealPads` and use `allPads` for excactly the combination you're looking for (both real devices and simulators) ... This change cannot be done lightly though, since existing code would continue to work but with different semantics 🤔I still need to figure out the best way to go forward ...

I'll leave this open if that's okay for you, thanks again for working on this :)

**Comment by denisenepraunig** [comment]
Hello @artemnovichkov! A few days ago we have finally created the 2.0 release and I have taken over the maintenance of DeviceKit from Dennis. I will take a look at your pull request and discuss it with @Zandor300 - our main contributor - on how to proceed further. Many thanks in advance for your time and effort!

File: Tests/Tests.swift [hunk 1]
```diff
@@ -381,6 +381,11 @@ class DeviceKitTests: XCTestCase {
     XCTAssertFalse(UIDevice.current.isBatteryMonitoringEnabled)
   }
 
+  func testSimilatorExtension() {
```
Comment (Zandor300) [inline Tests/Tests.swift]:
```suggestion
  func testSimulatorExtension() {
```

File: Device+Simulators.swift [hunk 2]
```diff
@@ -0,0 +1,16 @@
+//
+//  Copyright © 2018 Artem Novichkov. All rights reserved.
```
Comment (Zandor300) [inline Device+Simulators.swift]:
```suggestion
//===----------------------------------------------------------------------===//
//
// This source file is part of the DeviceKit open source project
//
// Copyright © 2014 - 2019 Dennis Weissmann and the DeviceKit project authors
//
// License: https://github.com/dennisweissmann/DeviceKit/blob/master/LICENSE
// Contributors: https://github.com/dennisweissmann/DeviceKit#contributors
//
//===----------------------------------------------------------------------===//
```

File: Device+Simulators.swift [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+//
+// This source file is part of the DeviceKit open source project
+//
+// Copyright © 2014 - 2019 Dennis Weissmann and the DeviceKit project authors
+//
+// License: https://github.com/dennisweissmann/DeviceKit/blob/master/LICENSE
+// Contributors: https://github.com/dennisweissmann/DeviceKit#contributors
+//
+//===----------------------------------------------------------------------===//
+//
```
Comment (Zandor300) [inline Device+Simulators.swift]:
```suggestion
```

File: Device+Simulators.swift [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
```
Comment (Zandor300) [inline Device+Simulators.swift]:
```suggestion
```

File: Device+Simulators.swift [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+public extension Sequence where Iterator.Element == Device {
+
+    var simulators: [Device] {
+        return map(Device.simulator)
+    }
+
+    var andSimulators: [Device] {
+        return self + simulators
+    }
```
Comment (Zandor300) [inline Device+Simulators.swift]:
```suggestion
    }

```

**Pull request closed without merging**